js/register.js

```diff
@@ -0,0 +1,46 @@
+import { fetcher } from "./fetcher.js";
+import { REGISTER_API_URL } from "./common/constants.js";
+
+// Register user
+// 1. Get the user data form the form
+// 2. Do a post request to the server
+// 3. User register only gets 201 created once, then error
+
+const form = document.querySelector("#registrationForm");
+const name = document.querySelector("#registerUsername");
+const email = document.querySelector("#registerEmail");
+const password = document.querySelector("#registerPassword");
+
+async function registerUser(user) {
+  console.log("Register user:", user); //User = user object
+  // POST request
+  const postBody = JSON.stringify(user);
```

Comment:
The call to fetcher should be in a try/catch block like in loginUser. Display a message to the user (in html) if the request was a success or a failure.